Helper/ShipmentOptions.php

```diff
@@ -82,23 +82,32 @@ public function getInsurance(): int
      */
     public function hasSignature(): bool
     {
-        return $this->optionIsEnabled(self::SIGNATURE);
+        $signatureFromOptions  = self::getValueOfOptionWhenSet(self::SIGNATURE, $this->options);
+        $signatureFromSettings = $this->optionIsEnabled(self::SIGNATURE);
+
+        return $signatureFromOptions ?? $signatureFromSettings;
     }
 
     /**
      * @return bool
      */
     public function hasOnlyRecipient(): bool
     {
-        return $this->optionIsEnabled(self::ONLY_RECIPIENT);
+        $onlyRecipientFromOptions  = self::getValueOfOptionWhenSet(self::ONLY_RECIPIENT, $this->options);
+        $onlyRecipientFromSettings = $this->optionIsEnabled(self::ONLY_RECIPIENT);
```

Comment:
De naam van deze variabele matcht niet echt met de naam van de functie die aangeroepen wordt, wat gebeurt hier nou echt?

En je mag van mij ook best dit schrijven:

```php
return $onlyRecipientFromOptions ?? $this->optionIsEnabled(self::ONLY_RECIPIENT);
```

Niet al te lang of onleesbaar en zorgt er wel voor dat als `$onlyRecipientFromOptions` niet null is dat `$optionIsEnabled` helemaal niet aangeroepen wordt